### PR TITLE
Fix compilation error in code for reset state variables with time type

### LINKED ISSUE #1938: Reset state fails with time type
The following program fails to compile, apparently because the reset state t has type `time`.  If you change its type to `int`, everything works.  The error message is:

```
/Users/eal/git/lf-3pi-template/src-gen/ModalBug/ModalBug.c:85:98: error: cannot take the address of an rvalue of type 'long long'
        envs[modalbug_main].modes->state_resets[modal_state_reset_count[modalbug_main]].source = &SEC(0);
                                                                                                 ^~~~~~~
1 error generated.
gmake[2]: *** [CMakeFiles/ModalBug.dir/build.make:118: CMakeFiles/ModalBug.dir/ModalBug.c.o] Error 1
```

Note the erroneous `&SEC(0)`.

```
target C
reactor C {
  input trigger:bool
  reset state t:time = 0s
  reaction(trigger) {=
    lf_print("t = %d", self->t);
    self->t = lf_time_logica();
  =}
}
main reactor {
  timer t(0, 1s)
  initial mode A {
    c = new C()
    reaction(t) -> reset(B), c.trigger {=
      lf_print("In A");
      lf_set(c.trigger, true);
      lf_set_mode(B);
    =}
  }
  mode B {
    reaction(t) -> reset(A) {=
      lf_print("In B");
      lf_set_mode(A);
    =}
  }
}
```

### PR DESCRIPTION
**Pull request by a-sr**
Closes #1938

### CODE REVIEW

File: test/C/src/modal_models/ResetStateVariableOfTypeTime.lf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    =}
+  }
+  
+  mode B {
+    reaction(t) -> reset(A) {=
+      lf_print("In B");
+      lf_set_mode(A);
+    =}
+  }
+}
```
Comment (edwardalee) [inline test/C/src/modal_models/ResetStateVariableOfTypeTime.lf:45]:
```suggestion
}

```

File: test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    =}
+  }
+  
+  mode B {
+    reaction(t) -> reset(A) {=
+      lf_print("In B");
+      lf_set_mode(A);
+    =}
+  }
+}
```
Comment (edwardalee) [inline test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf:45]:
```suggestion
}

```

**Review by edwardalee — APPROVED** [review]
LGTM!  Modulo formatting the tests, this seems to solve the problem. Thanks!

File: test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf (hunk 2 above)
Comment (a-sr) [reply test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf]: The formatting checks in CI to not permit an additional newline here. [error](https://github.com/lf-lang/lingua-franca/actions/runs/5960799155/job/16168815215#step:4:3028)

**Pull request merged**